docs/Makefile

```diff
@@ -53,3 +53,25 @@ html: cpp-apidoc spark-apidoc
 				--quiet
 	@echo
 	@echo "Build finished. The HTML pages are in $(BUILDDIR)/html."
+
+.PHONY: pyspark-apidoc
+pyspark-apidoc:
+	cd $(ROOTDIR)/pyspark && \
+	poetry run sphinx-apidoc -o $(ROOTDIR)/docs/pyspark/api graphar_pyspark/
+
+.PHONY: html-poetry
+html-poetry:
+	cd $(ROOTDIR)/pyspark && \
+	poetry run bash -c "cd $(ROOTDIR)/docs && $(SPHINXBUILD) -b html $(ALLSPHINXOPTS) $(BUILDDIR)/html"
```

Comment:
Ditto.

Reply:
The problem here is how sphinx works with python projects. Just because it is "native" for python, it may be used for generation of API docs. BUT, to do it, one need to have prject and dependencies in PYTHON_PATH... The simplest way to achieve it is to use the same dependency-management system, like for the project. I added to `pyproject.toml` dependencies from [docs/requirements.txt](https://github.com/alibaba/GraphAr/blob/main/docs/requirements.txt) and it works well... My recommendation is just to use poetry for docs generation (there is a github action for poetry and also for poetry cache, that allows also to improve building speed)

Reply:
The recommendation make sense to me, thanks for the explanation.